ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala

```diff
@@ -41,10 +42,21 @@ private[ftp] trait SftpOperations { _: FtpLike[SSHClient, SftpSettings] =>
 
     ssh.connect(host.getHostAddress, port)
 
-    if (credentials.password != "" && sftpIdentity.isEmpty)
-      ssh.authPassword(credentials.username, credentials.password)
+    if (credentials.password != "" && sftpIdentity.isDefined) {
```

Comment:
Try and use 
```
    sftpIdentity match {
      case Some(identity) if credentials.password != "" =>
```
to get a more readable structure.

Not introduced by you, but `credentials.password != ""` could mean it is `null`?!

Reply:
fixed

Reply:
What is fixed?
I'd still prefer the match clause with guards and extractor over the ifs.